### PR TITLE
Coherent naming: batchIndex -> batchId, using uint32 

### PR DESCRIPTION
**Pull request by josojo**
Coherent naming:  batchIndex->batchId, 

This PR makes sure that batchId is always used as uint32

---

testplan: e2e test

### CODE REVIEW

**Review by bh2smith — APPROVED** [review]
Sweet!

**Review by anxolin — CHANGES_REQUESTED** [review]
But I thought it was the other way around. The good one is not `batchId`?

Note that also the getter is called `getBatchId()`

**Comment by fleupold** [comment]
I slightly prefer batchIndex (but no strong feeling)

**Comment by josojo** [comment]
@anxolin 
while it is certainly an identifier, it is also an index, as it is constantly increasing.
I would suggest using index, as it is more explicit.

Would you be okay with it, once I change the getter to getBatchIndex as well?

**Comment by anxolin** [comment]
Well, In reality I could argue it's an identifier, since it identifies an user order. I agree it's also an index (it's the implementation details)

It's like in a auto increment field in a database we call it "index" instead of "id" cause it's autoincremented and it's and index

Having said that, I don't care that much if we want to call it index or id.

**Comment by anxolin** [comment]
What is more important for me is to be coherent, and write same concept always in the same way. Otherwise it's confusing I you see batchId, batchIndex, auctionNumber, auctionId, index, epoc,  etc..

**Comment by josojo** [comment]
Anxo you are right.

`ID`  puts the emphasis that it is a unique identifier and `index` is a little bit more explicit on how it is used. 

I would also feel comfortable with both decisions. 
I changed the code to index for now. ( Here and in the other PR)

**Comment by bh2smith** [comment]
I also prefer index since it could, technically, have started at zero (and IDs always start at 1).

**Review by anxolin — APPROVED** [review]
I approve cause I don't want to block this PR for this. But still like more `batchId` than `batchIndex` (on top of what I mentioned, it's shorter :) )

Also, changing to batchIndex affects more other modules since you are changing the interface, so that's more work

I think we should agree and try to do what's best here. We are reviewing specifically this in this PR, so I wouldn't do "batchIndex" for now and later we'll see. Now is the time to have consensus and leave it coherent

**Comment by fleupold** [comment]
@anxolin I don't understand what you mean with the last two sentences you wrote. On the one hand you say you would change it later, on the other hand you say now is the time to make it coherent.

I wouldn't be too concerned about "breaking" contract changes at this point (e.g. the method rename). In fact, I'd say now is the time to do it - It won't get easier. 
It would be easier if all the code was in one repo, so all we'd have to do is a global "find-replace", instead of updating all our micro-repos with different PRs at different times.

**Comment by anxolin** [comment]
@fleupold , sorry with last I tried to refer to Alex comment of:

> I changed the code to index for now

...but I think I wrote to fast, sorry for that. I was referring to his "for now". So I meant the opposite, we should do the change we believe is best now, instead of do something and later see

My vote is to call `orderId` and `batchId`, since they are used as identifiers, and it looks simpler to me. Besides, we already had this discussion and we agreed on id:
* https://github.com/gnosis/dex-contracts/pull/206
* https://github.com/gnosis/dex-contracts/pull/381

I approved the PR cause I don't find this discussion specifically important for me, so I'm ok if you all like calling it "index". I'll give us slight more work cause we need to apply the refactor in our code too though, so pls consider.

**Comment by nlordell** [comment]
I think that we should just choose one and run with it and I don't think this is super important. That being said, I do agree with @anxolin and prefer `batchId` over `batchIndex` as I would argue that the `batchWhatever` is not an index and batches aren't indexable. Instead, it is a derived value that uniquely identifies a given batch.
Personally I think that the best way to describe this is `batchNumber`.

**Review by nlordell — APPROVED** [review]

File: contracts/EpochTokenLocker.sol [hunk 1]
```diff
@@ -21,7 +21,7 @@ contract EpochTokenLocker {
     mapping(address => mapping(address => BalanceState)) private balanceStates;
 
     // user => token => lastCreditBatchId
-    mapping(address => mapping(address => uint256)) public lastCreditBatchId;
```
Comment (bh2smith) [inline contracts/EpochTokenLocker.sol:24]: Nice catch!

**Pull request merged**